### ISSUE TITLE
Feedback on lesson

### ISSUE THREAD

**Issue by weaverbel** [GULibrarySandbox/intro-computational-thinking#19]
I have a few comments. First, it is well written, it is easy to read through and not too hard to follow. Kudos. I actually finished it and expected to read on before I realised that was the end. Perhaps it is worth writing a bit more. Perhaps work through a few examples step by step so readers can gain confidence in what they’re doing. One small niggle is that usually introduce the concept of value assignment (e.g. x = 2) early on before branching or loop… about the same time as you introduce the topic “abstraction” actually. (Later on, I raise the topic of ambiguity on whether x = 2 means assign the value of 2 to x or test whether x has the value 2. 

But that may be too much at this level.) which reminds me that the first example of the sum of numbers relies on what readers will take as a math “trick” rather than breaking the problem down computationally. 

Another issue I discern with the examples (which also has impact on what they they think computers do) is that there is no context setting of what can and cannot be done: can the lost hikers ring a rescue helicopter? (Is there mobile coverage?) do the hikers know how to make fire? With a fertile enough imagination, you can solve any problem. But that doesn’t translate well if users don’t appreciate the limitations of the computer: we can’t just download all Twitter data to analyse and we can’t just write code to translate from any human language to another. This may be obvious to us, but perhaps not to the readers. 

The Little Women example, for instance, assumes you’ve read the novel and know who Jo is. But many readers (especially International students) wouldn’t have. This doesn’t matter of course, but highlights what kind of assumed knowledge one should expect for computational thinking: should we assume readers know what sorting is? Should readers be familiar with the difference between `.doc` and `.txt`? These are interesting questions especially when I look at the glossary and spot terms that were not used in the lesson. Should they be there? Or should the lesson be expanded?

**Comment by jduckles** [comment]
> But that may be too much at this level.) which reminds me that the first example of the sum of numbers relies on what readers will take as a math “trick” rather than breaking the problem down computationally.

Yes, I was having a hard time calling the trick an abstraction too. It is, but it doesn't seem like it has a clear logical path. I wonder if the example in #20 would allow for talking about an abstraction. Abstractions are kind of fiddly in that they require re-thinking the problem in different ways until an abstraction can be found.
